lib/rules/wrap-iife.js

```diff
@@ -52,11 +54,51 @@ module.exports = {
                         functionExpressionWrapped = wrapped(node.callee);
 
                     if (!callExpressionWrapped && !functionExpressionWrapped) {
-                        context.report(node, "Wrap an immediate function invocation in parentheses.");
+                        context.report({
+                            node,
+                            message: "Wrap an immediate function invocation in parentheses.",
+                            fix(fixer) {
+                                const nodeToSurround = style === "inside" ? node.callee : node;
+
+                                return fixer.replaceText(nodeToSurround, `(${sourceCode.getText(nodeToSurround)})`);
+                            }
+                        });
                     } else if (style === "inside" && !functionExpressionWrapped) {
-                        context.report(node, "Wrap only the function expression in parens.");
+                        context.report({
+                            node,
+                            message: "Wrap only the function expression in parens.",
+                            fix(fixer) {
+
+                                // The outer call expression will always be wrapped at this point.
+                                // Replace the range between the end of the function expression and the end of the call expression.
+                                // for example, in `(function(foo) {}(bar))`, the range `(bar))` should get replaced with `)(bar)`.
+                                // Replace the parens from the outer expression, and parenthesize the function expression.
```

Comment:
In our [coding conventions](http://eslint.org/docs/developer-guide/code-conventions), we prefer multi-line comments to be formatted like this:

```js
/*
 * First line of comment here.
 * Second line. Note that each line starts with asterisk aligned with the open-comment asterisk character.
 * Both comment start and end should be on their own line.
 * Comment ends with asterisk of close-comment token aligned with other asterisks.
 */
```